### ISSUE TITLE
Openseadragon and animated gifs

### ISSUE THREAD

**Issue by Salitehkat** [openseadragon/openseadragon#1497]
I was wondering if its posible to append animated gifs. I tried changing the jpg´s for gif´s in this [example](https://bl.ocks.org/fabiovalse/7705d5acb1cdd8678d39a74031e07580) and it does load the gif but static (no animation).  Also tried to find examples but nothing. Thanks a lot.

**Comment by Salitehkat** [comment]
Thanks for such a clear explanation. I am so new in this language that i feel helpless for creating a plug in. Hopefully i will start to understand more and will be able to collaborate. OSD is great for animated story maps, with custom images, svg, and animated gifs (for example and when possible). 
An alternative (meanwhile) solution could be to split the gif into single frames then create a spritesheet and animate it copying current frame.? read it in a post but not sure it can be applied in OSD. Many thanks for all this support.

**Comment by Salitehkat** [comment]
Edit: Or svg animations? Ill give a try and come back with results.

**Comment by iangilman** [comment]
Yes, there are a lot of directions you could go, depending upon what sort of effect you want. You might play with the various overlay plugins. The canvas overlay pretty much allows you to do whatever kind of animation you want, but you have to take responsibility for making it all happen. And yes, you might be able to do some interesting animation with the SVG overlay. 

You can use the built in OSD overlays to attach an animated GIF, but they don't have scaling support. It might be interesting to try creating a new HTML overlay that does include a `transform` for scaling, which would allow animated GIFs to zoom and pan in sync with the viewer.

Another option is multiple images in the viewer that you flip through (kind of like a spritesheet, but different pictures) to create an animation effect. I suppose you could even literally do a spritesheet in a single image, using the clipping feature to show just the area you want to. One challenge with that is managing preload. 

Anyway... lots of things to explore! I look forward to hearing what you find. :)

**Comment by antonioxdias** [comment]
While the implementation could vastly vary depending on your desired result, I made you a simple example of how you could go about solving this:
https://codepen.io/anon/pen/ePreVZ?editors=1010

As this could take any html element (even react components if you get creative), your other issue #1501 could also be solved this way.
I've also been using, for personal use, svg paths drawn with paperjs to get coordinates to re position my media elements.
I don't really think this is worthy of a plug-in, but if anyone deems necessary i can look into make a proper one....

**Comment by iangilman** [comment]
@antonioxdias That's beautiful! I've gone ahead and added it to my collection of OpenSeadragon Recipes: 

https://codepen.io/collection/APoyjJ/

I had to fork it, since I couldn't add an anonymous pen:

https://codepen.io/iangilman/pen/PyegxP 

If you want to make a non-anonymous version, I'm happy to use that instead.

I do think it would make a great plugin; I know it's not a lot of code, but there are lots of people who would feel more comfortable using it wrapped up in a neat plugin package.

Anyway, thank you for sharing this!

**Comment by iangilman** [comment]
Great! The new CodePen looks good. Would you mind giving it a title? It's currently just "Untitled". You could call it "OpenSeadragon GIF Overlay" or something. Also, feel free to remove my Patreon comment from the top of the JavaScript, since it's not my code :)

I agree that basing transformations off of an element's center is often better for this stuff. That said, many people are used to using the upper left. Anyway, whatever works best! I look forward to seeing the plugin.

**Comment by iangilman** [comment]
Thanks for naming the CodePen... I've added it to the collection :)
